File: BussinessLayer/Slipways.Domain/Models/Slipway.cs
using System;
using System.Collections.Generic;

namespace com.b_velop.Slipways.Domain.Models
{
    public class Slipway 
    {
        public Guid Id { get; set; }
        public DateTime Created { get; set; }
        public DateTime? Updated { get; set; }
        public double Latitude { get; set; }
        public double Longitude { get; set; }
        public string Name { get; set; }

        public int Rating { get; set; }
        public string Comment { get; set; }
        public string Street { get; set; }
        public string Postalcode { get; set; }
        public string City { get; set; }
        public string Country { get; set; }
        public decimal Costs { get; set; }
        public string Pro { get; set; }
        public string Contra { get; set; }

        public Guid? MarinaId { get; set; }
        public virtual Marina Marina { get; set; }
        public Guid WaterId { get; set; }
        public virtual Water Water { get; set; }
        public virtual ICollection<SlipwayExtra> SlipwayExtras { get; set; }
    }
}

File: Web/client-app/src/app/models/slipway.ts
export interface ISlipway {
    id: string;
    name: string;
    postalCode: string;
    city: string;
    street: string;
    country: string;
    latitude: number;
    longitude: number;
    water: string;
}
export interface ISlipwayFormValues extends Partial<ISlipway> {
    created?: Date;
}

export class SlipwayFormValues implements ISlipwayFormValues {
    id?: string = undefined;
    name: string = '';
    water: string = '';
    comment: string = '';
    created?: Date = undefined;
    city: string = '';
    country: string = '';
    street: string = '';
    postalCode: string = '';
    latitude: number = 0;
    longitude: number = 0;
    pro: string = '';
    contra: string = '';
    costs: number = -1;
    rating: number = 0;
    camping: boolean = false;
    parking: boolean = false;
    pier: boolean = true;

    constructor(init?: ISlipwayFormValues) {
        Object.assign(this, init);
    }
}
